### PR TITLE
feat(Guild): updates for Community guilds

### PR DESCRIPTION
**Pull request by advaith1**
**Please describe the changes this PR makes and why it should be merged:**
expands on #3810

- Public guilds are being renamed to Communities
- add preferredLocale property to Guild
- rulesChannel and publicUpdatesChannel can be set for any guild via API (not just public/community guilds)
- add support for setting rulesChannel, publicUpdatesChannel, and preferredLocale
- Previews can now be fetched for all guilds the bot is in and all Discoverable guilds

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/Guild.js [hunk 1]
```diff
@@ -1188,6 +1201,51 @@ class Guild extends Base {
     return this.edit({ banner: await DataResolver.resolveImage(banner), reason });
   }
 
+  /**
+   * Edits the rules channel of the guild.
+   * @param {ChannelResolvable} rulesChannel The new rules channel
+   * @param {string} [reason] Reason for changing the guild's rules channel
+   * @returns {Promise<Guild>}
+   * @example
+   * // Edit the guild system channel
```
Comment (SpaceEEC) [inline src/structures/Guild.js:1210]:
```suggestion
   * // Edit the guild rules channel
```

File: src/structures/Guild.js [hunk 2]
```diff
@@ -1188,6 +1201,51 @@ class Guild extends Base {
...
+    return this.edit({ rulesChannel }, reason);
+  }
+
+  /**
+   * Edits the public updates channel of the guild.
+   * @param {ChannelResolvable} publicUpdatesChannel The new public updates channel
+   * @param {string} [reason] Reason for changing the guild's public updates channel
+   * @returns {Promise<Guild>}
+   * @example
+   * // Edit the guild system channel
```
Comment (SpaceEEC) [inline src/structures/Guild.js:1225]:
```suggestion
   * // Edit the guild public updates channel
```

File: src/structures/Guild.js [hunk 3]
```diff
@@ -1188,6 +1201,51 @@ class Guild extends Base {
...
+    return this.edit({ publicUpdatesChannel }, reason);
+  }
+
+  /**
+   * Edits the preferred locale of the guild.
+   * @param {string} preferredLocale The new preferred locale of the guild
+   * @param {string} [reason] Reason for changing the guild's preferred locale
+   * @returns {Promise<Guild>}
+   * @example
+   * // Edit the guild name
```
Comment (SpaceEEC) [inline src/structures/Guild.js:1240]:
```suggestion
   * // Edit the guild preferred locale
```

**Comment by ShawnCZek** [comment]
As far as I know, `PUBLIC` and `PUBLIC_DISABLED` features can be dropped as they are not used by the API anymore.

**Pull request merged**